### ISSUE TITLE
Upgrade from 1.6.1 to 1.7.1 fails with DatabaseException

### ISSUE THREAD

**Issue by bratao** [LINBIT/linstor-server#153]
Hi,

I have a cluster that uses a etcd database with a 1.6.1 version. I tried to upgrade today to the latest version but I got this problem:
```
com.linbit.linstor.InitializationException: Initial load from the database failed

	at com.linbit.linstor.core.DbDataInitializer.initialize(DbDataInitializer.java:121)

	at com.linbit.linstor.core.ApplicationLifecycleManager.startSystemServices(ApplicationLifecycleManager.java:87)

	at com.linbit.linstor.core.Controller.start(Controller.java:296)

	at com.linbit.linstor.core.Controller.main(Controller.java:505)

Caused by: com.linbit.linstor.dbdrivers.DatabaseException: Failed to deserialize json array. Table: RESOURCE_GROUPS, column: POOL_NAME

	at com.linbit.linstor.dbdrivers.AbsDatabaseDriver$RawParameters.getAsStringList(AbsDatabaseDriver.java:302)

	at com.linbit.linstor.core.objects.ResourceGroupDbDriver.load(ResourceGroupDbDriver.java:303)

	at com.linbit.linstor.core.objects.ResourceGroupDbDriver.load(ResourceGroupDbDriver.java:57)

	at com.linbit.linstor.dbdrivers.etcd.ETCDEngine.loadAll(ETCDEngine.java:143)

	at com.linbit.linstor.dbdrivers.AbsDatabaseDriver.loadAll(AbsDatabaseDriver.java:100)

	at com.linbit.linstor.dbdrivers.DatabaseLoader.loadAll(DatabaseLoader.java:247)

	at com.linbit.linstor.core.DbDataInitializer.loadCoreObjects(DbDataInitializer.java:176)

	at com.linbit.linstor.core.DbDataInitializer.initialize(DbDataInitializer.java:108)

	... 3 more

Caused by: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'DfltStorPool': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')

 at [Source: (String)"DfltStorPool"; line: 1, column: 25]

	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1840)

	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:722)

	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._reportInvalidToken(ReaderBasedJsonParser.java:2867)

	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleOddValue(ReaderBasedJsonParser.java:1913)

	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:772)

	at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:4340)

	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4189)

	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3205)

	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3173)

	at com.linbit.linstor.dbdrivers.AbsDatabaseDriver$RawParameters.getAsStringList(AbsDatabaseDriver.java:283)

	... 10 more

com.linbit.linstor.dbdrivers.DatabaseException: Failed to deserialize json array. Table: RESOURCE_GROUPS, column: POOL_NAME

	at com.linbit.linstor.dbdrivers.AbsDatabaseDriver$RawParameters.getAsStringList(AbsDatabaseDriver.java:302)

	at com.linbit.linstor.core.objects.ResourceGroupDbDriver.load(ResourceGroupDbDriver.java:303)

	at com.linbit.linstor.core.objects.ResourceGroupDbDriver.load(ResourceGroupDbDriver.java:57)

	at com.linbit.linstor.dbdrivers.etcd.ETCDEngine.loadAll(ETCDEngine.java:143)

	at com.linbit.linstor.dbdrivers.AbsDatabaseDriver.loadAll(AbsDatabaseDriver.java:100)

	at com.linbit.linstor.dbdrivers.DatabaseLoader.loadAll(DatabaseLoader.java:247)

	at com.linbit.linstor.core.DbDataInitializer.loadCoreObjects(DbDataInitializer.java:176)

	at com.linbit.linstor.core.DbDataInitializer.initialize(DbDataInitializer.java:108)

	at com.linbit.linstor.core.ApplicationLifecycleManager.startSystemServices(ApplicationLifecycleManager.java:87)

	at com.linbit.linstor.core.Controller.start(Controller.java:296)

	at com.linbit.linstor.core.Controller.main(Controller.java:505)

Caused by: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'DfltStorPool': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')

 at [Source: (String)"DfltStorPool"; line: 1, column: 25]

	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1840)

	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:722)

	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._reportInvalidToken(ReaderBasedJsonParser.java:2867)

	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleOddValue(ReaderBasedJsonParser.java:1913)

	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:772)

	at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:4340)

	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4189)

	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3205)

	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3173)

	at com.linbit.linstor.dbdrivers.AbsDatabaseDriver$RawParameters.getAsStringList(AbsDatabaseDriver.java:283)

	... 10 more

com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'DfltStorPool': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')

 at [Source: (String)"DfltStorPool"; line: 1, column: 25]

	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1840)

	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:722)

	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._reportInvalidToken(ReaderBasedJsonParser.java:2867)

	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleOddValue(ReaderBasedJsonParser.java:1913)

	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:772)

	at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:4340)

	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4189)

	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3205)

	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3173)

	at com.linbit.linstor.dbdrivers.AbsDatabaseDriver$RawParameters.getAsStringList(AbsDatabaseDriver.java:283)

	at com.linbit.linstor.core.objects.ResourceGroupDbDriver.load(ResourceGroupDbDriver.java:303)

	at com.linbit.linstor.core.objects.ResourceGroupDbDriver.load(ResourceGroupDbDriver.java:57)

	at com.linbit.linstor.dbdrivers.etcd.ETCDEngine.loadAll(ETCDEngine.java:143)

	at com.linbit.linstor.dbdrivers.AbsDatabaseDriver.loadAll(AbsDatabaseDriver.java:100)

	at com.linbit.linstor.dbdrivers.DatabaseLoader.loadAll(DatabaseLoader.java:247)

	at com.linbit.linstor.core.DbDataInitializer.loadCoreObjects(DbDataInitializer.java:176)

	at com.linbit.linstor.core.DbDataInitializer.initialize(DbDataInitializer.java:108)

	at com.linbit.linstor.core.ApplicationLifecycleManager.startSystemServices(ApplicationLifecycleManager.java:87)

	at com.linbit.linstor.core.Controller.start(Controller.java:296)

	at com.linbit.linstor.core.Controller.main(Controller.java:505)

17:45:56.702 [Main] ERROR LINSTOR/Controller - SYSTEM - Initial load from the database failed [Report number 5ED9334C-00000-000000]
```
Preventing me to start my linstor cluster. If I revert to the older version, everything works good!

**Comment by bratao** [comment]
Here is a etcd dump if it helps to debug the problem
[etcd-dump.zip](https://github.com/LINBIT/linstor-server/files/4731847/etcd-dump.zip)

**Comment by rp-** [comment]
Sorry for the late reply, but we had a lot of work and some vacations.

We forgot to migrate a etcd field with this update, you can work around for now by transforming your resource groups from strings into json lists.

```
export ETCDCTL_API=3
# e.g. get all you resource groups with storage pools set:
etcdctl get --prefix /LINSTOR/RESOURCE_GROUPS/ | grep POOL_NAME -A 1
# now put them again as json lists e.g.:
etcdctl put /LINSTOR/RESOURCE_GROUPS/MYGRP/POOL_NAME '["DfltStorPool"]'
...
```

The next version will have a automatic migration for that.

**Comment by rp-** [comment]
fixed in 1.7.2
